cmake/utils.cmake

```diff
@@ -148,18 +148,37 @@ function(mayaUsd_install_rpath rpathRef NAME)
 endfunction()
 
 function(mayaUsd_promoteMayaUsdHeader)
-    set(srcFile ${CMAKE_CURRENT_SOURCE_DIR}/base/mayaUsd.h.src)
+    set(srcFile ${CMAKE_CURRENT_SOURCE_DIR}/mayaUsd.h.src)
     set(dstFile ${CMAKE_BINARY_DIR}/include/mayaUsd/mayaUsd.h)
     if (NOT EXISTS ${dstFile})
         message(STATUS "promoting: " ${srcFile})
     endif()
     configure_file(${srcFile} ${dstFile})
 endfunction()
 
-function(mayaUsd_promoteHeaderList)
-    foreach(header ${ARGV})
+function(mayaUsd_promoteHeaderList target)
+    cmake_parse_arguments(PREFIX
+        "TARGET"
```

Comment:
From the doc, the first arg is the options. "target" is already a arg passed to the function. I had already made this change in my local branch like so:
```cmake
    cmake_parse_arguments(PREFIX 
        ""          # options
        "SUBDIR"    # one_value keywords
        "FILES"     # multi_value keywords
        ${ARGN}
    )
```
Why is DIRNAME mandatory? 

I had also fixed the copyDirectory below.

Reply:
@seando-adsk 
Right, I didn't really have to pass the target and was solely added for debugging purposes.  Target is not really used inside the function.

Both DIRNAME  and HEADERS keywords are mandatory. DIRNAME is needed because we no longer have the header files with the subdir prefix ( e.g `base/api.h`, `base/debugCodes.h`, etc... ). Instead the subdir name now needs to be passed to mayaUsd_promoteHeaderList function as a separate argument.

```
e.g
set(headers
    api.h
    debugCodes.h
)

I am going to change get rid of the target and rename DIRNAME to SUBDIR. 
e.g
mayaUsd_promoteHeaderList(HEADERS ${headers} SUBDIR base)
mayaUsd_promoteHeaderList(HEADERS ${headers} SUBDIR nodes)
mayaUsd_promoteHeaderList(HEADERS ${headers} SUBDIR listeners)
```

Did you remove target from mayaUsd_copyDirectory? I believe target was needed to be passed to `add_custom_command` command?

Reply:
@HamedSabri-adsk Okay I see for both those keywords being mandatory now. No I didn't remove TARGET from any of the functions.